nbuild/stdenv/install.py

```diff
@@ -21,43 +22,56 @@ def make_symlink(src, dst):
     os.symlink(src, dst)
 
 
-def make_cp(*src, dest, root='', args=''):
+def make_cp(*files, dest, root=''):
     package = current_build().current_package
-    for filename in src:
-        path = f'{package.install_dir}/{root}/{filename}/'
-        cmd(f'cp {args} {path} {dest}')
+    for filename in files:
+        path = f'{package.install_dir}/{root}/{filename}'
+        shutil.copy2(path, dest)
 
 
-def make_mkdir(dir, args=''):
+def make_mkdir(dir):
     package = current_build().current_package
     path = f'{package.install_dir}/{dir}/'
-    cmd(f'mkdir {args} {path}')
+    if not os.path.exists(path):
+        os.makedirs(path)
 
 
-def make_rm(*files, root='', args=''):
+def make_rm_file(*files, root=''):
     package = current_build().current_package
     for filename in files:
-        path = f'{package.install_dir}/{root}/{filename}/'
-        cmd(f'rm {args} {path}')
+        path = f'{package.install_dir}/{root}/{filename}'
+        os.remove(path)
 
 
-def make_mv(*src, dest, root='', args=''):
+def make_rm_directory(dir, root=''):
     package = current_build().current_package
-    for filename in src:
-        path = f'{package.install_dir}/{root}/{filename}/'
-        cmd(f'mv {args} {path} {dest}')
+    path = f'{package.install_dir}/{root}/{dir}'
+    os.rmdir(path)
+
+
+def make_mv(*files, dest, root=''):
+    package = current_build().current_package
+    for filename in files:
+        path = f'{package.install_dir}/{root}/{filename}'
+        shutil.move(path, dest)
 
 
-def make_chmod(dest, args):
+def make_chmod(dest, mode, root=''):
     package = current_build().current_package
-    path = f'{package.install_dir}/{dest}/'
-    cmd(f'chmod {args} {path}')
+    path = f'{package.install_dir}/{root}/{dest}'
+    octal_mode = "0o" + mode
+    os.chmod(path, octal_mode)
 
 
-def make_sed(regex, filename, args=''):
+def make_sed(regex, replacement, filename, root=''):
     package = current_build().current_package
-    path = f'{package.install_dir}/{filename}/'
-    cmd(f'sed {args} {regex} {path}')
+    path = f'{package.install_dir}/{root}/{filename}'
+    input = open(path, 'r')
```

Comment:
I'm not really sure about not calling cmd('sed') here. For the other functions it makes sense to use existing python functions because they are usually just wrappers on sycalls, but here it's a fully working binary with many options. I'd say it's ok for now, but I wouldn't be surprised if that needed to change afterwards.

Anyways, it is good practice to use the `with` statement in those cases, as the file would be automatically closed at the end of the scope.

```python
with open(path, 'r') as input:
    content = input.read()
with open(path, 'w') as f:
    f.write(re.sub(regex, replacement, content))
```

Reply:
Yeah, for this one it's fine i guess